### ISSUE TITLE
Feedback on Syncing.md

### ISSUE THREAD

**Issue by frrist** [filecoin-project/specs#82]
## What is chain syncing in Filecoin
- Initially I am given the impression that a node runs the "syncing" protocols for sometime, and then stops once it has downloaded the entire chain from the network. However, we state that the protocol is used to sync blocks the nodes themselves have mined. I believe this means "syncing" is a never ending/long running process, if so this should be explicitly stated.
- A few sentences on why we need to do this, and what happens if we don't do this would be nice to have.

## Interface
I understand we haven't written the actual interface for this, but some code comments would be really nice here. Basically make this pass a linter -- leave a summary at the top of each method stating what it does, and why it's there.

## Syncing Mode
- Is the goal of `syncing` mode to download the entire chain state from the network and ensure its valid? If so, let's say something like that explicitly.
- While the following is correct:
> In this mode of operation a filecoin node should not mine or send messages as it will not be able to successfully generate heaviest blocks or reference the correct state of the chain to verify that messages will execute as expected.

We should also state that the node will be unable to wait on messages (`go-filecoin message wait`) and will be unable to check wallet balances (`go-filecoin wallet balance <address>`). Currently, only once the chain has been fully synced can the node receive funds and wait on messages (just manually verified this on commit 29325c33caaa484ebe395fc5d26fd876cbdb86c3).  

 
## Caught Up Mode
- What is the goal of the `caught up` protocol? Sounds like it collects all valid tipsets for the EC round? Also sounds like it updates the nodes heaviest tipset, does this mean its always running?
- This leads to the question: Will the node ever leave this state? For example, it is stated that it exits when an error is encountered, when does it re-enter this state?
- What does `Punctual` mean here?

Overall, it sounds like this part of the protocol is overloaded and closely coupled with expected consensus, I find this part of the syncing spec to be the most confusing part. 

## Dependencies
- The `Punctual` method is fuzzy to me, needs a glossary entry.

**Comment by pooja** [comment]
@whyrusleeping To check that all feedback was addressed

**Comment by mhammersley** [comment]
@sternhenri for awareness

**Comment by sternhenri** [comment]
Sorry for lag. @frrist, this is a great question. I feel like a similar concern was raised by @gnunicorn and addressed in https://github.com/filecoin-project/specs/pull/295. This should be reflected in current spec.

Was it addressed to your satisfaction? If not, what remains? I'll revise the spec accordingly.

**Comment by frrist** [comment]
Since this issue was created the spec has been modified to describe the syncing process at a higher level and most of the implementation details have been removed. This alleviates almost all the concerns laid out above as the details are left to the implementer -- this works for me :+1:. 

Last bit that remains unaddressed: Will a definition be given for Punctuality, or do we no longer require one? This was [brought up](https://github.com/filecoin-project/specs/pull/295#issuecomment-501014795) during review of #295.

**Comment by sternhenri** [comment]
Punctuality no longer appears in any section of the spec. I believe we can assume it will not be defined/used. Will leave it to @whyrusleeping and @pooja  to reopen this issue if this is a mistake (and sorry if so).

**Comment by anorth** [comment]
For clarity, at present go-filecoin has a "finality window" of 600 blocks and, after initially syncing to a head discovered by secure bootstrapping, the syncer will reject blocks farther than that window earlier than or later the best known head. Without such a window, a node can be forced to download essentially arbitrarily long fake chains.

The trusted bootstrap process may be re-executed in order to discover a true head outside that window. Is that acceptable?

**Comment by sternhenri** [comment]
That seems good to me. To rephrase, I think you said
- Reject blocks without a common prefix with current best head within finality window, unless re running bootstrap.

That said I didn't fully parse
> the syncer will reject blocks farther than that window earlier than or later the best known head

I think there's room for further optimizations splitting how you treat blocks at nearby heights (checking for common prefix within finality window) vs blocks from far in the future (which I believe you can reject way before 600 blocks away given you likely should have heard of them within 30-50 rounds at most)... That said, for now if it works, I think treating both the same is correct.
